pkg/syncer/specsyncer.go

```diff
@@ -77,58 +76,49 @@ func NewSpecSyncer(from, to *rest.Config, syncedResourceTypes []string, clusterI
 	}
 	fromClient := fromClients.Cluster(logicalClusterID)
 	toClient := dynamic.NewForConfigOrDie(to)
-	return New(fromDiscovery, fromClient, toClient, upsertIntoDownstream, deleteFromDownstream, func(c *Controller, gvr schema.GroupVersionResource) cache.ResourceEventHandlerFuncs {
-		return cache.ResourceEventHandlerFuncs{
-			AddFunc: func(obj interface{}) { c.AddToQueue(gvr, obj) },
-			UpdateFunc: func(oldObj, newObj interface{}) {
-				if !deepEqualApartFromStatus(oldObj, newObj) {
-					c.AddToQueue(gvr, newObj)
-				}
-			},
-			DeleteFunc: func(obj interface{}) { c.AddToQueue(gvr, obj) },
-		}
-	}, syncedResourceTypes, clusterID)
-}
-
-// TODO:
-// This function is there as a quick and dirty implementation of namespace creation.
-// In fact We should also be getting notifications about namespaces created upstream and be creating downstream equivalents.
-func (c *Controller) ensureNamespaceExists(namespace string) error {
```

Comment:
@ncdc Maybe just revert this removal vs reimplementing?

Reply:
K